core/src/main/java/org/apache/iceberg/util/ThreadPools.java

```diff
@@ -61,6 +62,26 @@ public static ExecutorService getWorkerPool() {
     return WORKER_POOL;
   }
 
+  public static ExecutorService newWorkerPool(String namePrefix, Integer parallelism) {
+    return MoreExecutors.getExitingExecutorService(
+        (ThreadPoolExecutor) Executors.newFixedThreadPool(
+            Optional.ofNullable(parallelism).orElse(WORKER_THREAD_POOL_SIZE),
```

Comment:
should we make the param a primitive type and provide an overload method without the poolSize param?

Reply:
I'm okay either way.